### ISSUE TITLE
Missing Gateway

### ISSUE THREAD

**Issue by mtuska** [jbowdre/phpIPAM-for-vRA8#3]
Attempting to sync IPAM with vRA while some subnets have no gateway results in an error:
```
[2021-12-14 14:34:59,806] [INFO] - Querying for auth credentials
[2021-12-14 14:34:59,839] [INFO] - Credentials obtained successfully!
[2021-12-14 14:35:00,169] [INFO] - Found subnet: 0.0.0.0/28 - XYZ.
[2021-12-14 14:35:00,263] [INFO] - Found subnet: 0.0.0.0/27 - XYZ.
[2021-12-14 14:35:00,368] [INFO] - Found subnet: 0.0.0.0/23 - XYZ.
[2021-12-14 14:35:00,473] [INFO] - Found subnet: 0.0.0.0/24 - XYZ.
Traceback (most recent call last):
  File "abx_wrapper.py", line 325, in main
    response = exec_function(wrapperCtx, request, f)
  File "abx_wrapper.py", line 223, in exec_function
    funcResult = handle(ctx, req)
  File "/function/source.py", line 42, in handler
    return ipam.get_ip_ranges()
  File "/function/vra_ipam_utils/ipam.py", line 64, in get_ip_ranges
    result = self.do_get_ip_ranges(auth_credentials, cert)
  File "/function/source.py", line 88, in do_get_ip_ranges
    gateway = gw_req.json()['data'][0]['ip']
KeyError: 'data'
```

It seems that it will still return 200 since the request was valid, but with no data. For get_ip_ranges/source.py I suggest
`if gw_req.status_code == 200 and not "data" in gw_req.json():`

Additionally, it would be nice to be able to specify the filter_by rule. Currently it's forced as `filter_by=isPool&filter_value=1` but, since I believe you can add custom values to the vRA integration page, you could add filter_by and filter_value fields. Then instead of having these subnets be the only 'pools' in phpIPAM, I could create a custom field called vRA as a boolean, and simply filter for that. `?filter_by=custom_vRA&filter_value=1`

**Comment by jbowdre** [comment]
Hey @mtuska, thanks for the report. I see what you mean on both points. 

I'll try to get both implemented soonish, though it probably won't be until next week at the earliest (thanks, log4j!).

**Comment by jbowdre** [comment]
I published a new release [v1.0.6](https://github.com/jbowdre/phpIPAM-for-vRA8/releases/tag/v1.0.6) with this fix and enhancement. 

Thanks again for the report!

**Comment by jbowdre** [comment]
@mtuska  And I'm now realizing that I completely misread your suggestion on the filter fields. I'll take another swing at this and see what I can come up with.

**Comment by jbowdre** [comment]
@mtuska Okay, I think [this](https://github.com/jbowdre/phpIPAM-for-vRA8/releases/tag/v1.0.7) is closer to what you were after.
![Screenshot 2021-12-21 21 58 43](https://user-images.githubusercontent.com/61015723/147035288-61a96227-6b04-4dd1-976b-46f4e3af879f.png)

**Comment by mtuska** [comment]
The new filter is great, thank you. Apparently both you and I overlooked an issue I made in the other direction for the gateway... woops! It has been adjusted to `if gw_req.status_code == 200 and not 'data' in gw_req.json():` which in fact will only try to pull the gateway data if there is no data. Technically, it should be `if gw_req.status_code == 200 and 'data' in gw_req.json():` without the not. Again, my bad on providing that, even my local tests of the script had it correct I just typed it out wrong.
